User can choose language

ViktorHek opened the pull request:
[User can choose language](https://www.pivotaltracker.com/story/show/176031749)
```
As a user
In order to be able to read articles in my native language,
I would like to switch between languages English and Swedish
```
## Changes proposed in this pull request:
* Visitor can change language to English, swedish or cat-language (a.k.a "Meow")
* Adding Icons that, when clicked, change the language of the page
## What I have learned working on this feature:
* how to change the language for test created in the frontend 
## Packages/Gems added
* i18next
* i18next-xhr-backend
* i18next-browser-languagedetector
## Screenshots (Client)
![Screenshot from 2021-01-09 19-25-29](https://user-images.githubusercontent.com/72013716/104105744-88eeb880-52b0-11eb-9e7f-85f85ef91938.png)

ViktorHek committed:
adding translation packege. setting up translation. creating a tranlation to Global Politics

ViktorHek committed:
adding contry flag with onClick, fixing typo

ViktorHek committed:
language change now workes

ViktorHek committed:
adding translating for remaning test. adding cat language

ViktorHek committed:
adding test to test if user is able to change language

salindholm committed:
Fixing typo and adding translation to home-link

emtalen commented on src/i18n.js:
```diff
@@ -0,0 +1,19 @@
+  .init({
+    fallbackLng: ["en"],
+    whitelist: ["en", "sv", "cat"],
+    debug: true,
```
you can remove the `debug: true`, it's not needed (sorry for messy demo)

ViktorHek replied:
fixed. it's removed :)

emtalen commented on src/index.js:
```diff
@@ -8,18 +8,22 @@ import { BrowserRouter } from "react-router-dom";
 
 ReactDOM.render(
-  <Provider store={store}>
+  <Suspense fallback={<div>Loading.... Mauw</div>}>
```
the `Suspense` should only wrap the `App`

ViktorHek replied:
we have moved it inside Provider, but still outside of BrowserRouter

ViktorHek committed:
removing unnececary code in i18n.js. moving Suspense inside of Provider in index.js

ViktorHek committed:
Merge branch 'visitor_can_change_languege' of https://github.com/ViktorHek/kitty_news_client_user into visitor_can_change_languege

tochman commented on src/components/Header.jsx:
```diff
@@ -1,11 +1,15 @@
+import { Menu, Segment, Flag, Icon } from "semantic-ui-react";
 import { useDispatch } from "react-redux";
 import { getArticles } from "../modules/getArticles";
+import i18n from "../i18n";
```
I would import this module in `index.js` that is the highest level. Please try that and let me know if you are okay

ViktorHek replied:
we tried but then our tests failed. it said i18n is not defined in the header component

tochman replied:
I think that is fixable with 
```
const { t, I18n } = useTranslation();
```

tochman replied:
or perhaps: 
```
const { t, i18n } = useTranslation();
```

Note the lowercased `i` 

Or a combination thereof...

KyunginNa replied:
The second one worked! Thanks

KyunginNa committed:
Refactoring configureStore
Changing import of i18n in Header Component to make it imported only in index.js
